integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonFileMetastore.scala

```diff
@@ -155,9 +153,12 @@ class CarbonFileMetastore extends CarbonMetaStore {
           case Some(name) if name.equals("org.apache.spark.sql.CarbonSource") => name
           case _ => throw new NoSuchTableException(database, tableIdentifier.table)
         }
-        new CarbonSource().createRelation(sparkSession.sqlContext,
-          catalogTable.storage.properties
-        ).asInstanceOf[CarbonDatasourceHadoopRelation].carbonRelation
+        val identifier: AbsoluteTableIdentifier = AbsoluteTableIdentifier.from(
```

Comment:
This code is added to prevent infinite loop which occured due to TablePath changes. @ravipesala  Please have a look.